app/views/questionnaire.py

```diff
@@ -219,40 +210,125 @@ def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unu
                                  survey_title=TemplateRenderer.safe_content(schema['title']))
 
 
+@questionnaire_blueprint.route('<group_id>/<int:group_instance>/permanent-or-family-home', methods=["POST"])
+@login_required
+def post_everyone_at_address_confirmation(eq_id, form_type, collection_id, group_id, group_instance):
+    if request.form.get('permanent-or-family-home-answer') == 'No':
+        _remove_repeating_on_household_answers(get_answer_store(current_user), group_id)
+    return post_block(eq_id, form_type, collection_id, group_id, group_instance, 'permanent-or-family-home')
+
+
+def _render_block(block, current_location, post_form=None):
+
+    if post_form is None:
+        context = _get_context(block, current_location, get_answer_store(current_user))
+    else:
+        context = {'form': post_form, 'block': block}
+
+    full_routing_path = path_finder.get_routing_path()
+
+    return _build_template(
+        current_location,
+        context,
+        block['type'],
+        routing_path=full_routing_path)
+
+
+def _is_valid_location(location):
+    return location in path_finder.get_routing_path(
+        location.group_id,
+        location.group_instance,
+    )
+
+
+def _is_valid_group(group_id):
+    return group_id in SchemaHelper.get_group_ids(g.schema_json)
+
+
+def _generate_wtf_form(form, block, location):
+    error_messages = SchemaHelper.get_messages(g.schema_json)
+    disable_mandatory = 'action[save_sign_out]' in form
+
+    wtf_form, _ = post_form_for_location(
+        block,
+        location,
+        get_answer_store(current_user),
+        request.form,
+        error_messages,
+        disable_mandatory)
+
+    return wtf_form
+
+
+def _submit_answers(eq_id, form_type, collection_id):
+    metadata = get_metadata(current_user)
+    answer_store = get_answer_store(current_user)
+
+    return submit_answers(eq_id, form_type, collection_id, metadata, answer_store)
+
+
+def _next_location_url(location):
+    metadata = get_metadata(current_user)
+    return location.url(metadata)
+
+
+def _is_end_of_questionnaire(block, location):
+    end_blocks = ["Summary", "Confirmation"]
+
+    return location is None and \
```

Comment:
It's a little confusing as it reads like `location` would be the location of the block, it would be clearer named `next_location`